docs/Dockerfile

```diff
@@ -0,0 +1,6 @@
+ARG  BRANCH=master
+FROM quay.io/nmstate/kubernetes-nmstate-docs:${BRANCH}
+COPY Makefile Gemfile* /docs/
+RUN make -C docs install
+COPY . /docs/
+RUN make -C docs check
```

Comment:
We don't have check anymore. And why do we call it while building the container?

Reply:
We do have `check` target and depends on `build`, Dockerfile is consumed by the prow automation that checks links are broken and the like with `html-proofer`

Reply:
Usually, Dockerfile is used to build a container image containing needed tools and only then the build image is used to use the tool.

Is not this the case here? Are we making the check as a part of the container building?

Reply:
This docker file contains the build and the webpage, and is based on the same container published so we do incremental builds.

I can move the webpage build out of the Dockerfile but I think it does not worth it.

One of the cool things is that you can start the page without the repo just `docker run ... make serve`

Reply:
You got me by the `make serve`. All good then.